src/Microsoft.ML.Data/DataLoadSave/Binary/BinaryLoader.cs

```diff
@@ -755,8 +752,9 @@ private static VersionInfo GetVersionInfo()
                 modelSignature: "BINLOADR",
                 //verWrittenCur: 0x00010001, // Initial
                 //verWrittenCur: 0x00010002, // Generated row index column
-                verWrittenCur: 0x00010003, // Number of blocks to put in the shuffle pool
-                verReadableCur: 0x00010003,
+                //verWrittenCur: 0x00010003, // Number of blocks to put in the shuffle pool
+                verWrittenCur: 0x00010004, // Row index column no longer being generated
```

Comment:
>0x00010004 [](start = 31, length = 10)

Please add a unit test loading an old model to see that the backwards compatibility works.

Reply:
Hi @yaeldekel , I added a unit test where I mirrored EntryPointPipelineEnsemble(), but instead of saving and then loading that model, I load a model saved with BinaryLoader.GetVersionInfo().verWrittenCur = 0x00010003. Let me know what you think!

Reply:
Hi @yaeldekel , there is indeed already a test that is checking for compatibility with BinaryLoader.GetVersionInfo().verWrittenCur = 0x00010003, which is the unit test `LoadOriginalBinaryLoaderModel`. As I am updating the BinaryLoader's written version from 0x00010003 to 0x00010004, I don't see the need to add a complicated new test for backwards compatibility, since a simpler version of a test like this already exists. :D

Reply:
I am not sure I follow this. Since you are deleting a variable from the saved model, don't you need a test to validate that?

---
In reply to: [398764732](https://github.com/dotnet/machinelearning/pull/4714#discussion_r398764732) [](ancestors = 398764732)

Reply:
The test that Mustafa mentioned uses a model that was pre-generated using version 0x00010003, which has the row index column saved in the model, and it validates that this model can be used with the current code (version 0x00010004). Other tests (I hope they exist :-)) that create a binary loader, serialize and deserialize it test that the version 0x00010004 serialization is valid. I would be very surprised if there is no test that does this, but @mstfbl, can you verify this, just in case?

---
In reply to: [403399348](https://github.com/dotnet/machinelearning/pull/4714#discussion_r403399348) [](ancestors = 403399348,398764732)

Reply:
Hi @yaeldekel and @harishsk , yup there are tests such as `EntryPointPipelineEnsemble` and `EntryPointMulticlassPipelineEnsemble` that do test BinaryLoader for the current 0x00010004 version.